colour/io/ocio.py

```diff
@@ -0,0 +1,110 @@
+
+
+@required('OpenColorIO')
+def process_image_OpenColorIO(a, *args, **kwargs):
+    """
+    Processes given image with *OpenColorIO*.
+
+    Parameters
+    ----------
+    a : array_like
+        Image to process with *OpenColorIO*.
+
+    Other Parameters
+    ----------------
+    \\*args : list, optional
+        Arguments for `Config.getProcessor` method.
+        See https://opencolorio.readthedocs.io/en/latest/api/config.html for
+        more information.
+    config : unicode, optional
+        *OpenColorIO* config to use for processing.
+    use_cpu : bool, optional
+        Whether to use the *CPU* or *GPU* processor.
+
+    Returns
+    -------
+    ndarray
+        Processed image.
+
+    Examples
+    --------
+    >>> import os
+    >>> import PyOpenColorIO as ocio
+    >>> from colour.utilities import full
+    >>> config = os.path.join(
+    ...     os.path.dirname(__file__), 'tests', 'resources',
+    ...     'config-aces-reference.ocio.yaml')
+    >>> a = full([4, 2, 3], 0.18)
+    >>> process_image_OpenColorIO(  # doctest: +ELLIPSIS
+    ...     a, 'ACES - ACES2065-1', 'ACES - ACEScct', config=config)
+    array([[[ 0.4135878...,  0.4135878...,  0.4135878...],
+            [ 0.4135878...,  0.4135878...,  0.4135878...]],
+    <BLANKLINE>
+           [[ 0.4135878...,  0.4135878...,  0.4135878...],
+            [ 0.4135878...,  0.4135878...,  0.4135878...]],
+    <BLANKLINE>
+           [[ 0.4135878...,  0.4135878...,  0.4135878...],
+            [ 0.4135878...,  0.4135878...,  0.4135878...]],
+    <BLANKLINE>
+           [[ 0.4135878...,  0.4135878...,  0.4135878...],
+            [ 0.4135878...,  0.4135878...,  0.4135878...]]], dtype=float32)
+    >>> process_image_OpenColorIO(  # doctest: +ELLIPSIS
+    ...     a, 'ACES - ACES2065-1', 'Display - sRGB',
+    ...     'Output - SDR Video - ACES 1.0', ocio.TRANSFORM_DIR_FORWARD,
+    ...     config=config)
+    array([[[ 0.3559523...,  0.3559525...,  0.3559525...],
+            [ 0.3559523...,  0.3559525...,  0.3559525...]],
+    <BLANKLINE>
+           [[ 0.3559523...,  0.3559525...,  0.3559525...],
+            [ 0.3559523...,  0.3559525...,  0.3559525...]],
+    <BLANKLINE>
+           [[ 0.3559523...,  0.3559525...,  0.3559525...],
+            [ 0.3559523...,  0.3559525...,  0.3559525...]],
+    <BLANKLINE>
+           [[ 0.3559523...,  0.3559525...,  0.3559525...],
+            [ 0.3559523...,  0.3559525...,  0.3559525...]]], dtype=float32)
+    """
+
+    import PyOpenColorIO as ocio
+
+    use_cpu = kwargs.get('use_cpu', True)
+
+    config = kwargs.get('config')
+    config = (ocio.Config.CreateFromEnv()
+              if config is None else ocio.Config.CreateFromFile(config))
+
+    a = np.atleast_3d(as_float_array(a, dtype=np.float32))
+    height, width, channels = a.shape
+
+    processor = config.getProcessor(*args)
+
+    processor = (processor.getDefaultCPUProcessor()
+                 if use_cpu else processor.getDefaultGPUProcessor())
```

Comment:
I don't think you can use a GPU Processor like you can a CPU Processor; i.e., there's no "apply" method....

Reply:
I actually haven't tried nor looked at it! I will dig through the OCIO codebase.

Reply:
So looks like it is more involved indeed: https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/0021372b4dba3444e21a36d846b327efad054f8d/tests/python/GpuShaderDescTest.py#L127

Reply:
And on top of that you need to create an OpenGL context, manually upload any texture data needed and execute your draw call...!

Reply:
Yeah, I saw that, will try to update that tonight, be great to consider making that easier, there is probably no good reasons why it cannot be abstracted for a user, c.f. `import cupy as np`.

Reply:
I went path-of-least-resistance and removed the non-working GPU path, we can figure that out in a later PR.

Reply:
So, confirming that we will not have GPU processing anytime soon: https://github.com/AcademySoftwareFoundation/OpenColorIO/pull/1493/commits/4a8dac4affef4fdea3c2c5b4d67abde7c803c645